google-analytics-query.html

```diff
@@ -208,105 +240,137 @@
          * @attribute fields
          * @type string
          */
-
-        observe: {
-          ids: 'getData',
-          startDate: 'getData',
-          endDate: 'getData',
-          metrics: 'getData',
-          dimensions: 'getData',
-          sort: 'getData',
-          filters: 'getData',
-          segment: 'getData',
-          samplingLevel: 'getData',
-          startIndex: 'getData',
-          maxResults: 'getData',
-          output: 'getData',
-          fields: 'getData'
-        },
-
-        userAuthorized: function() {
-          this.super();
-          this.getData();
-        },
-
-        userUnauthorized: function() {
-          this.super();
-          this.data = null;
+        fields: {
+          type: String,
+          value: ''
         },
 
         /**
-         * Query the Google Analytics Core Reporting API.
-         *
-         * @method getData
+         * Some legacy variable: I believe the GA team was using _src for analytics purposes.
+         * @attribute underscoreSrc
+         * @type string
          */
-        getData: function() {
-          if (this.authorized && this.hasRequiredParams) {
-
-            // Required parameters.
-            var query = {
-              'ids': this.ids,
-              'start-date': this.startDate,
-              'end-date': this.endDate,
-              'metrics': this.metrics
-            };
-
-            // Optional parameters.
-            if (this.dimensions) query.dimensions = this.dimensions;
-            if (this.sort) query.sort = this.sort;
-            if (this.filters) query.filters = this.filters;
-            if (this.segment) query.segment = this.segment;
-            if (this.samplingLevel) query.samplingLevel = this.samplingLevel;
-            if (this.startIndex) query['start-index'] = this.startIndex;
-            if (this.maxResults) query['max-results'] = this.maxResults;
-            if (this.output) query.output = this.output;
-            if (this.fields) query.fields = this.fields;
-
-            this.beforeQuerySend(query);
-
-            gapi.client.analytics.data.ga.get(query)
-                .execute(this.handleResponse.bind(this));
-
-            return true;
-          }
+        underscoreSrc: {
```

Comment:
I struggled with this one. In the original code, the default was 'gwc-ga-query', and chart was overriding it with 'gwc-ga-chart'. Can I just hardcode this one?